api/src/auth/drivers/oauth2.ts

```diff
@@ -254,6 +255,14 @@ export function createOAuth2AuthRouter(providerName: string): Router {
 		respond
 	);
 
+	router.post(
+		'/callback',
+		bodyParser.urlencoded({ extended: false }),
+		(req, res) => {
+			res.redirect(303, `./callback?${new URLSearchParams(req.body)}`);
```

Comment:
303 ensures the redirect will be GET. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/303